code/implementation/docker/common/src/main/java/io/cattle/platform/docker/transform/DockerTransformerImpl.java

```diff
@@ -266,7 +266,7 @@ void setLogConfig(Instance instance, Map<String, Object> fromInspect) {
     void setLabels(Instance instance, Map<String, Object> fromInspect) {
         // Labels not yet implemented in docker-java. Need to use the raw map
         Object l = CollectionUtils.getNestedValue(fromInspect, "Config", "Labels");
-        Map<String, String> cleanedLabels = new HashMap<String, String>();
+        Map<String, Object> cleanedLabels = new HashMap<String, Object>();
```

Comment:
@ibuildthecloud This is related to adding the `io.rancher.container.volume_cleanup_strategy=none` for native containers [in ContainerEventCreate](https://github.com/rancher/cattle/pull/1228/files#diff-c5617822afbbc645620f4bd13d6caff8R196).

When I add the label there, when this code finally gets called as part of back population, it will clobber any existing labels that didn't come back as part of the docker inspect.